.github/workflows/pathogen-repo-build.yaml

```diff
@@ -212,7 +212,7 @@ jobs:
         name: Get AWS Batch job id
         id: aws-batch
         run: |
-          echo "AWS_BATCH_JOB_ID=$(tail -n1 build.log | sed -nE 's/.+attach ([-a-f0-9]+).+/\1/p')" >> "$GITHUB_ENV"
+          echo "AWS_BATCH_JOB_ID=$(sed -nE 's/.+AWS Batch Job ID\:.+ ([-a-f0-9]+)$/\1/p' < build.log)" >> "$GITHUB_ENV"
```

Comment:
Does this pattern actually match the output? I think the ANSI escapes (e.g. for bold) would get in the way here with this pattern as-is.

Reply:
Ahh, I see you [already ran into this](https://github.com/nextstrain/.github/issues/50#issuecomment-1709098826).

Reply:
I would think to future-proof this a bit more so it doesn't break when Nextstrain CLI eventually stops emitting ANSI control sequences to non-TTY stdouts.

Reply:
Yup, leaving as-is as a work-around until we get a more suitable output from Nextstrain CLI. 

> Longer term, we could have Nextstrain CLI emit this data in a better way that's more suitable for programmatic usage (maybe piggybacking on https://github.com/nextstrain/cli/issues/113 or https://github.com/nextstrain/cli/issues/125). Would you be interested in doing that work?

Definitely interested in this work!

Reply:
Yay! Happy to chat more about it elsewhere, but some thoughts to get out of my head.

The data I'd think to emit would be the Batch job id, but also other info like the relevant S3 paths and maybe other bits and bots.

Some ways I've thought about emitting it in the past:

1. `nextstrain build` writes a JSON file when requested, e.g. by `--aws-batch-info job.json` (exact spelling TBD).
2. `nextstrain build` emits delimited YAML in its normal output, always, which can be easily extracted and parsed, e.g.

    Submitting job
    AWS Batch Job ID: 2ad078eb-630b-4ad2-80d0-bb898f5b6c48
    ---
    nextstrain-build-id: …
    aws-batch-job-id: …
    workdir-url: s3://…
    ...
    Watching job status

3. As part of https://github.com/nextstrain/cli/issues/113 and/or https://github.com/nextstrain/cli/issues/125, `nextstrain build` internally tracks job details and a new command provides subsequent access to them, e.g. `nextstrain build-info most-recent` or something.

(1) probably makes the most sense, especially as a starting point. (3) might be where we go eventually, but there's a lot more design considerations that we probably don't want/need to tackle now. (2) is kinda nice in some ways but annoying in others. And it's accomplishable with (1) anyway (well, assuming there's a YAML output option).

Reply:
This should probably become an issue over in the right repo…